_posts/2021-10-26-bwc-testing-for-opensearch.markdown

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title:  "Backwards Compatibility Testing for OpenSearch"
+authors: 
+  - vachshah
+  - vemsarat
+date: 2021-10-26 01:01:01 -0700
+categories: 
+  - technical-post
+twittercard:
+  description: "This post provides details on the framework in OpenSearch for backwards compatibility testing"
+---
+
+Backwards Compatibility (BWC) testing is used to test and determine the safe upgrade paths from a supported BWC version to the current version. The framework in OpenSearch allows to run these BWC tests for all supported BWC versions in OpenSearch allowing safe upgrade paths between versions. This framework is now extended to work for plugins and they can introduce their BWC tests without creating individual frameworks of their own. This post provides details on the framework in OpenSearch for backwards compatibility.
+
+## The Framework
+
+As a general idea for BWC tests, the framework supports to spin up a test cluster with a supported BWC version, then upgrade the nodes to the current version and test various features and functionalities as a result of the upgrade. This allows to test the compatibility of the code between versions.
```

Comment:
'the framework supports **spinning** up a test cluster'
'This allows **for testing** the'